tests/app/conftest.py

```diff
@@ -1555,7 +1555,7 @@ def _sample_notification_history(
             key_type=key_type,
             api_key=api_key,
             api_key_id=api_key.id,
-            reference=reference,
+            reference=reference or uuid4(),
```

Comment:
This looks like JavaScript.  Please use the Pythonic syntax, which is admittedly more verbose in this case.
```suggestion
            reference=reference if (reference is not None) else uuid4(),
```
Are you sure this shouldn't be `str(uuid4())`?

Reply:
The field in the model is a db.string but it doesn't hurt to be safe.